packages/backend-deployer/src/cdk_deployer_singleton_factory.ts

```diff
@@ -1,19 +1,30 @@
 import {
+  AmplifyIOHost,
   BackendIdentifier,
-  type PackageManagerController,
+  PackageManagerController,
+  SDKProfileResolver,
 } from '@aws-amplify/plugin-types';
 import { CDKDeployer } from './cdk_deployer.js';
 import { CdkErrorMapper } from './cdk_error_mapper.js';
 import { BackendLocator } from '@aws-amplify/platform-core';
 import { BackendDeployerOutputFormatter } from './types.js';
+import { Toolkit } from '@aws-cdk/toolkit-lib';
 
 export type DeployProps = {
   secretLastUpdated?: Date;
   validateAppSources?: boolean;
+  /**
+   * @deprecated CDK toolkit now accepts profile only in the constructor instead of at runtime. This param will be removed in next MV
+   * Use `sdkProfileResolver` in `BackendDeployerFactory` instead to set the profile
+   */
```

Comment:
if we do MV then we can just remove. (and below as well).

Reply:
These props are coming from `sandbox` and `cli` package, I don't think we want to do MV for them just yet.

Reply:
https://github.com/aws-amplify/amplify-backend/pull/2503/files#diff-19a3328a512ce2869b263ce83233f5c1c00f3c4506ed966a8fab5a4f204ab3dfR41-R44

Reply:
Updated the sandbox to have MV and removed profile options.